_posts/2024-01-16-owasp-top10-with-abstract-examples-part-2.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: post
+title: OWASP top ten 2021 explained with non-technical examples, </br>Part 2
+author: petteri.poyhtari
+excerpt: >
+   How could we better make the most typical risks of web applications aware of all project personnel, in order to improve the quality of the entire web infrastructure. Part 2 blog post about OWASP top 10 with physical world analogies.
+tags:
+ - DevSecOps
+ - InfoSec
+ - OWASP
+ - Quality
+ - Security
+ - Software security
+ - Web development
+---
+
+## What is this blog about?
+
+This is the second part of our [OWASP top 10](https://owasp.org/Top10/) explanation with non-technical examples. In [part 1](https://dev.solita.fi/2024/01/09/owasp-top10-with-abstract-examples-part-1.html) I lead into the subject why understanding the most common risks for non-technical persons is also important and explained risks from one to three. Let's continue with risks four to ten.
+
+## 4 Insecure Design
+
+Applications can be designed with care and safety in mind from the beginning or with the fastest possible return of investment in mind. Unfortunately, the latter mostly means that information security gives way to "more important features". [Insecure design](https://owasp.org/Top10/A04_2021-Insecure_Design/) means that security is not built in, but is added afterwards when the product is ready enough. This fails in most cases and is more expensive. Since it is more expensive to do, it is either hardly done or not done at all. For example, application layers or the network can be taken into account in the design. Fixing these later is expensive.
```

Comment:
"return *on* investment" lienee yleisempi muoto.

en ymmärrä tuota "For example, application layers or the network can be taken..." tarkoittaako tämä, että ne pitää huomioida, tai se tulee kalliiksi?

Reply:
Selvennystä tosiaan vaatii. Tässä haetaan softan arkkitehtuuria, niin voisin sen lisätä tuohon. Vaikka näin:
"For example, the application architecture, how system and network layers are implemented, must be considered in the design."